index.html

```diff
@@ -0,0 +1,191 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title>
+      Direct Sockets API
+    </title>
+    <script src="https://www.w3.org/Tools/respec/respec-w3c" class="remove"
+    defer></script>
+    <script class='remove'>
+      var respecConfig = {
+        specStatus: "unofficial",
+        github: {
+          repoURL: "WICG/raw-sockets",
+          branch: "main"
+        },
+        group: "wicg",
+        editors: [
+        {
+          name: "Eric Willigers",
+          company: "Google Inc.",
+          companyURL: "https://google.com",
+          w3cid: 67534
+        }],
+        xref: "web-platform",
+        // eventually add mdn: true, and caniuse: "direct-sockets"
+      };
+    </script>
+  </head>
+  <body data-cite="Direct Sockets">
+    <section id="abstract">
+      <p>
+        This specification defines an API that allows web applications to talk
+        to servers and devices that have their own protocols incompatible with
+        what’s available on the web.
+      </p>
+    </section>
+    <section>
+      <h2>
+        API definition
+      </h2>
+      <section data-dfn-for="Navigator">
+        <h3>
+          Extensions to the `Navigator` interface
+        </h3>
+        <pre class="idl">
+          partial interface Navigator {
+            [SecureContext] Promise&lt;TCPSocket&gt; openTCPSocket(optional SocketOptions options = {});
+            [SecureContext] Promise&lt;UDPSocket&gt; openUDPSocket(optional SocketOptions options = {});
+          };
+        </pre>
+        <p>
+          User agents that do not support direct sockets SHOULD NOT expose
+          {{Navigator/openTCPSocket()}} or {{Navigator/openUDPSocket()}} on the
+          {{Navigator}} interface.
+        </p>
+        <section data-dfn-for="SocketOptions">
+          <h3>
+            `SocketOptions` dictionary
+          </h3>
+          <pre class="idl">
+          dictionary SocketOptions {
+            DOMString localAddress;
+            unsigned short localPort;
+
+            DOMString remoteAddress;
+            unsigned short remotePort;
+
+            unsigned long sendBufferSize;
+            unsigned long receiveBufferSize;
+
+            boolean keepAlive;
+            boolean noDelay;
+          };
+        </pre>
+          <p>
+            The <dfn>SocketOptions</dfn> dictionary consists of several
+            optional members:
+          </p>
+          <dl data-sort="">
+            <dt>
+              <dfn>localAddress</dfn> member
```

Comment:
Add a little description for what each option does/means?

Reply:
done